### PR TITLE
feat(CI): turn docker images into locally produced matrix

### PR DESCRIPTION
**Pull request by oklopfer**
## Purpose

need both multi-arch (amd64+arm64) and multi-distro (ubuntu+debian) images. 

## Approach

Example workflow:
Screenshot 2024-02-22 at 1 39 04 PM

https://github.com/oklopfer/pacstall/actions/runs/8008403762

Screenshot 2024-02-22 at 1 46 32 PM

(yes they are the same commit, but on different branches; shows that develop does not mark latest)

Screenshot 2024-02-22 at 1 57 15 PM

(with version tag)

naming scheme:
`ghcr.io/pacstall/{ubuntu,debian}-{distro_release}:{ref}`

refs are:
- `master`, `master-{hash_shortform}`
- `develop`, `master-{hash_shortform}`
- `version_tag` (like `4.3.2`)
- `latest` (either `master` or `version_tag`, never `develop`)

architecture is automatic to host device

## Progress

- [x] fork my https://github.com/rhino-linux/docker program into https://github.com/pacstall/docker for storage of the dockerfile construction script (packages currently built over there, but to trigger builds on push, we move workflow here)
- [x] create a super dynamic workflow file that can run independent of the docker repository, as it can just curl in the above mentioned script
- [x] have it publish automatically on pushes to master + develop, and on tag creation
- [x] have backup: 
 Screenshot 2024-02-22 at 1 33 04 PM
 
- [x] can select branch to re-publish for master + develop, and can type in the box for version tags
- [x] yes, it does actually hard switch to the pacstall version
- [x] figure out how to name master + develop docker tags `${branch}-${git_hash}` instead of `${branch}-${YYYYMMDD}`

## Checklist

- [x] I confirm that I have read the [contributing guidelines](https://github.com/pacstall/pacstall/blob/develop/CONTRIBUTING.md), and this pull request is abiding by all the clauses stated in the guideline.

### CODE REVIEW

**Comment by oklopfer** [comment]
Idk it's acting weird rn

**Comment by oklopfer** [comment]
https://github.com/pacstall/docker/actions/runs/8007690263

idk its working here for some reason

File: .github/workflows/publish-docker.yml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+          fi
+          platform=${{ matrix.platform }}
+          arch=${platform#*/}
+          distro=${{ matrix.OS }}
+          distro_tag="${distro/:/-}"
+          curl -fsSL https://raw.githubusercontent.com/pacstall/docker/master/pacstall-docker-builder -o pacstall-docker-builder
+          chmod +x pacstall-docker-builder
+          ./pacstall-docker-builder -f -a ${arch} -d ${distro} -v ${vertag}
+          echo "BUILT_DOCKFILE=Dockerfile-Pacstall-${vertag}-${arch}-${distro_tag}-$(date +%Y%m%d)"  >> $GITHUB_ENV
+          echo "DOCK_LABEL=${vertag}"  >> $GITHUB_ENV
```
Comment (Elsie19) [inline .github/workflows/publish-docker.yml:54]:
```suggestion
          echo "DOCK_LABEL=${vertag}" >> $GITHUB_ENV
```

**Review by Elsie19 — CHANGES_REQUESTED** [review]

**Review by Elsie19 — APPROVED** [review]
LGTM 🚀 (I suppose)

**Pull request merged**